src/napari_micromanager/_mda_handler.py

```diff
@@ -132,11 +133,31 @@ def _on_mda_started(self, sequence: MDASequence) -> None:
         for i in self.viewer.layers:
             if i.metadata.get("uid") == sequence.uid:
                 self._mmc.mda.toggle_pause()
-                return
 
-    @ensure_main_thread  # type: ignore [misc]
+        self._mda_running = True
+        # init index will always be less than any event index
+        self._largest_idx: tuple[int, ...] = (-1,)
+        self._io_t = create_worker(self._watch_mda, _start_thread=True)
+
+    def _watch_mda(self) -> None:
+        print("starting to watch")
+        while self._mda_running:
+            try:
+                self._process_frame(*self._deck.pop())
+            except IndexError:
+                pass
+        import time
+
+        time.sleep(0.5)
```

Comment:
todo: Make a real choice for this number. I chose this based on what was convenient for development. Ideally it would be also be adjustable - open to suggestions.